packages/driver/src/cypress/cy.ts

```diff
@@ -69,9 +68,13 @@ const setTopOnError = function (Cypress, cy: $Cy) {
 
   curCy = cy
 
-  // prevent overriding top.onerror twice when loading more than one
-  // instance of test runner.
-  if (top.__alreadySetErrorHandlers__) {
+  try {
+    // prevent overriding top.onerror twice when loading more than one
+    // instance of test runner.
+    if (top.__alreadySetErrorHandlers__) {
+      return
+    }
+  } catch (err) {
```

Comment:
won't this catch cypress errors and prevent it from propagating up to the user?

Reply:
It only catches if the AUT is cross-origin, where referencing `top` will throw a cross-origin error.